apps/spruce/src/pages/waterfall/VersionLabel.tsx

```diff
@@ -0,0 +1,106 @@
+import styled from "@emotion/styled";
+import { Body, InlineCode } from "@leafygreen-ui/typography";
+import { Link } from "react-router-dom";
+import { Unpacked } from "@evg-ui/lib/types/utils";
+import { useProjectHealthAnalytics } from "analytics/projectHealth/useProjectHealthAnalytics";
+import { StyledRouterLink } from "components/styles";
+import { getVersionRoute, getTriggerRoute } from "constants/routes";
+import { WaterfallQuery } from "gql/generated/types";
+import { useSpruceConfig, useDateFormat } from "hooks";
+import { shortenGithash } from "utils/string";
+import { jiraLinkify } from "utils/string/jiraLinkify";
+
+type VersionFields = NonNullable<
+  Unpacked<WaterfallQuery["waterfall"]["versions"]>["version"]
+>;
+
+export const VersionLabel: React.FC<VersionFields> = ({
+  author,
+  createTime,
+  gitTags,
+  id,
+  message,
+  revision,
+  upstreamProject,
+}) => {
+  const getDateCopy = useDateFormat();
+  const createDate = new Date(createTime);
+
+  const spruceConfig = useSpruceConfig();
+  const jiraHost = spruceConfig?.jira?.host ?? "";
+
+  const { sendEvent } = useProjectHealthAnalytics({
+    page: "Waterfall",
+  });
+
+  return (
+    <VersionContainer>
+      <Body>
+        <InlineCode
+          as={Link}
+          onClick={() => {
+            sendEvent({
+              name: "Clicked commit label",
+              "commit.type": "active",
+              link: "githash",
+            });
+          }}
+          to={getVersionRoute(id)}
+        >
+          {shortenGithash(revision)}
+        </InlineCode>{" "}
+        {getDateCopy(createDate, { omitSeconds: true, omitTimezone: true })}
+      </Body>
+      {upstreamProject && (
+        <Body>
+          Triggered by:{" "}
+          <StyledRouterLink
+            onClick={() => {
+              sendEvent({
+                name: "Clicked commit label",
+                "commit.type": "active",
+                link: "upstream project",
+              });
+            }}
+            to={getTriggerRoute({
+              triggerType: upstreamProject.triggerType,
+              upstreamTask: upstreamProject.task,
+              upstreamVersion: upstreamProject.version,
+              upstreamRevision: upstreamProject.revision,
+              upstreamOwner: upstreamProject.owner,
+              upstreamRepo: upstreamProject.repo,
+            })}
+          >
+            {upstreamProject.project}
+          </StyledRouterLink>
+        </Body>
+      )}
+      {/* @ts-expect-error */}
+      <CommitMessage title={message}>
```

Comment:
I tried this out in prod and noticed some weird formatting I tried this out with prod data and noticed some weird formatting 
![image](https://github.com/user-attachments/assets/050dd3a9-2b94-4f0c-b1e2-165363587810)

Reply:
I think this is expected for now as we don't yet have logic to determine the number of columns to render on the page. I opted not to set a fixed width on each column as I don't think that benefits the layout. So the extra width seems ok at the moment! Unless there's something else you thought was weird

Reply:
Just wanted to make sure that the lines not wrapping for the 1st,3rd, and 5th commit were expected. But if its only temporary that sgtm!

Reply:
Ah, that's because the other commits have short messages and Git Tag is rendered on its own line so as not to be cut off 👍